bin/test-workflow/8_app_verify_authentication.sh

```diff
@@ -0,0 +1,183 @@
+bl_retry_constant "${RETRIES}" "${RETRY_WAIT}"  check_pods
+
+if [[ "$PLATFORM" == "openshift" ]]; then
+  echo "Waiting for deployments to become available"
+
+  check_deployment_status(){
+    # [[ "$(deployment_status "test-app-summon-init")" == "Complete" ]] &&
+    # [[ "$(deployment_status "test-app-with-host-outside-apps-branch-summon-init")" == "Complete" ]] &&
+    [[ "$(deployment_status "test-app-summon-sidecar")" == "Complete" ]] # &&
+    # [[ "$(deployment_status "test-app-secretless")" == "Complete" ]]
+  }
+  bl_retry_constant "${RETRIES}" "${RETRY_WAIT}"  check_deployment_status
+
+  sidecar_pod=$(get_pod_name test-app-summon-sidecar)
+  init_pod=$(get_pod_name test-app-summon-init)
+  init_pod_with_host_outside_apps=$(get_pod_name test-app-with-host-outside-apps-branch-summon-init)
+  secretless_pod=$(get_pod_name test-app-secretless)
+
+  # Routes are defined, but we need to do port-mapping to access them
+  oc port-forward "$sidecar_pod" 8081:8080 > /dev/null 2>&1 &
+  SIDECAR_PORT_FORWARD_PID=$!
+  oc port-forward "$init_pod" 8082:8080 > /dev/null 2>&1 &
+  INIT_PORT_FORWARD_PID=$!
+  oc port-forward "$secretless_pod" 8083:8080 > /dev/null 2>&1 &
+  SECRETLESS_PORT_FORWARD_PID=$!
+  oc port-forward "$init_pod_with_host_outside_apps" 8084:8080 > /dev/null 2>&1 &
+  INIT_WITH_HOST_OUTSIDE_APPS_PORT_FORWARD_PID=$!
+
+  curl_cmd=curl
+  sidecar_url="localhost:8081"
+  init_url="localhost:8082"
+  secretless_url="localhost:8083"
+  init_url_with_host_outside_apps="localhost:8084"
+else
+  if [[ "$TEST_APP_LOADBALANCER_SVCS" == "true" ]]; then
+    echo "Waiting for external IPs to become available"
+    check_services(){
+      # [[ -n "$(external_ip "test-app-summon-init")" ]] &&
+      # [[ -n "$(external_ip "test-app-with-host-outside-apps-branch-summon-init")" ]] &&
+      [[ -n "$(external_ip "test-app-summon-sidecar")" ]] # &&
+      # [[ -n "$(external_ip "test-app-secretless")" ]]
+    }
+    bl_retry_constant "${RETRIES}" "${RETRY_WAIT}"  check_services
+
+    curl_cmd=curl
+    init_url=$(external_ip test-app-summon-init):8080
+    init_url_with_host_outside_apps=$(external_ip test-app-with-host-outside-apps-branch-summon-init):8080
+    sidecar_url=$(external_ip test-app-summon-sidecar):8080
+    secretless_url=$(external_ip test-app-secretless):8080
+
+  else
+    # Apps don't have loadbalancer services, so test by curling from
+    # a pod that is inside the KinD cluster.
+    curl_cmd=pod_curl
+    init_url="test-app-summon-init.$TEST_APP_NAMESPACE_NAME.svc.cluster.local:8080"
+    init_url_with_host_outside_apps="test-app-with-host-outside-apps-branch-summon-init.$TEST_APP_NAMESPACE_NAME.svc.cluster.local:8080"
+    sidecar_url="test-app-summon-sidecar.$TEST_APP_NAMESPACE_NAME.svc.cluster.local:8080"
+    secretless_url="test-app-secretless.$TEST_APP_NAMESPACE_NAME.svc.cluster.local:8080"
+  fi
+fi
+
+echo "Waiting for urls to be ready"
+
+check_urls(){
+  (
+    # $curl_cmd -sS --connect-timeout 3 "$init_url" &&
```

Comment:
Thinking ahead to when we add more application integrations...
Ideally, these curl commands for each app integration should be run conditionally based on env variable.
However, if it's possible to move these calls into Helm tests that are defined for each subchart, then this would be a lot less fugly with conditionals scattered through this test code.

If we're not going to add conditionals (based on env variable enables) for the short term, we should at least file an issue or create a spike to either add conditional logic in, or figure out how to do this as Helm tests in each of the subcharts.

Reply:
Added #316